File: drestaurant-libs/drestaurant-courier/src/test/java/com/drestaurant/courier/domain/CourierOrderSagaTest.java
package com.drestaurant.courier.domain;

import org.axonframework.test.saga.FixtureConfiguration;
import org.axonframework.test.saga.SagaTestFixture;
import org.junit.Before;
import org.junit.Test;

import com.drestaurant.common.domain.model.AuditEntry;

public class CourierOrderSagaTest {

	private FixtureConfiguration testFixture;
	private AuditEntry auditEntry;
	private static final String WHO = "johndoe";
	private String orderId;
	private String courierId;

	@Before
	public void setUp() throws Exception {
		testFixture = new SagaTestFixture<>(CourierOrderSaga.class);
		auditEntry = new AuditEntry(WHO);
		orderId = "orderId";
		courierId = "courierId";
	}
	
	@Test
	public void courierOrderAssigningInitiatedTest() throws Exception {

		testFixture.givenNoPriorActivity()
		           .whenAggregate(orderId)
				   .publishes(new CourierOrderAssigningInitiatedEvent(courierId, orderId, auditEntry))
				   .expectActiveSagas(1)
				   .expectDispatchedCommands(new ValidateOrderByCourierCommand(orderId, courierId, auditEntry));
	}
	@Test
	public void courierNotFoundTest() throws Exception {

		testFixture.givenAggregate(orderId)
		           .published(new CourierOrderAssigningInitiatedEvent(courierId, orderId, auditEntry))
		           .whenPublishingA(new CourierNotFoundForOrderEvent (courierId, orderId, auditEntry))
		           .expectActiveSagas(0)
		           .expectDispatchedCommands(new MarkCourierOrderAsNotAssignedCommand(orderId, auditEntry));
	}
	@Test
	public void orderValidatedWithErrorByCourierTest() throws Exception {

		testFixture.givenAggregate(orderId)
		           .published(new CourierOrderAssigningInitiatedEvent(courierId, orderId, auditEntry))
		           .whenPublishingA(new OrderValidatedWithErrorByCourierEvent (courierId, orderId, auditEntry))
		           .expectActiveSagas(0)
		           .expectDispatchedCommands(new MarkCourierOrderAsNotAssignedCommand(orderId, auditEntry));
	}
	@Test
	public void orderValidatedWithSuccessByCourierTest() throws Exception {

		testFixture.givenAggregate(orderId)
		           .published(new CourierOrderAssigningInitiatedEvent(courierId, orderId, auditEntry))
		           .whenPublishingA(new OrderValidatedWithSuccessByCourierEvent (courierId, orderId, auditEntry))
		           .expectActiveSagas(0)
		           .expectDispatchedCommands(new MarkCourierOrderAsAssignedCommand(orderId, courierId, auditEntry));
	}
	
}

File: drestaurant-libs/drestaurant-courier/src/test/kotlin/com/drestaurant/courier/domain/CourierOrderSagaTest.kt
package com.drestaurant.courier.domain

import com.drestaurant.common.domain.api.model.AuditEntry
import com.drestaurant.courier.domain.api.model.CourierId
import com.drestaurant.courier.domain.api.model.CourierOrderId
import org.axonframework.test.saga.FixtureConfiguration
import org.axonframework.test.saga.SagaTestFixture
import org.junit.Before
import org.junit.Test
import java.util.*

class CourierOrderSagaTest {

    private lateinit var testFixture: FixtureConfiguration
    private lateinit var auditEntry: AuditEntry
    private lateinit var orderId: CourierOrderId
    private lateinit var courierId: CourierId
    private val who = "johndoe"

    @Before
    fun setUp() {
        testFixture = SagaTestFixture(CourierOrderSaga::class.java)
        auditEntry = AuditEntry(who, Calendar.getInstance().time)
        orderId = CourierOrderId("orderId")
        courierId = CourierId("courierId")
    }

    @Test
    fun courierOrderAssigningInitiatedTest2() {

        testFixture.givenNoPriorActivity()
                .whenAggregate(orderId.toString())
                .publishes(CourierOrderAssigningInitiatedInternalEvent(courierId, orderId, auditEntry))
                .expectActiveSagas(1)
                .expectDispatchedCommands(ValidateOrderByCourierInternalCommand(orderId, courierId, auditEntry))
    }

    @Test
    fun courierNotFoundTest() {

        testFixture.givenAggregate(orderId.toString())
                .published(CourierOrderAssigningInitiatedInternalEvent(courierId, orderId, auditEntry))
                .whenPublishingA(CourierNotFoundForOrderInternalEvent(courierId, orderId, auditEntry))
                .expectActiveSagas(0)
                .expectDispatchedCommands(MarkCourierOrderAsNotAssignedInternalCommand(orderId, auditEntry))
    }

    @Test
    fun orderValidatedWithErrorByCourierTest() {

        testFixture.givenAggregate(orderId.toString())
                .published(CourierOrderAssigningInitiatedInternalEvent(courierId, orderId, auditEntry))
                .whenPublishingA(CourierValidatedOrderWithErrorInternalEvent(courierId, orderId, auditEntry))
                .expectActiveSagas(0)
                .expectDispatchedCommands(MarkCourierOrderAsNotAssignedInternalCommand(orderId, auditEntry))
    }

    @Test
    fun orderValidatedWithSuccessByCourierTest() {

        testFixture.givenAggregate(orderId.toString())
                .published(CourierOrderAssigningInitiatedInternalEvent(courierId, orderId, auditEntry))
                .whenPublishingA(CourierValidatedOrderWithSuccessInternalEvent(courierId, orderId, auditEntry))
                .expectActiveSagas(0)
                .expectDispatchedCommands(MarkCourierOrderAsAssignedInternalCommand(orderId, courierId, auditEntry))
    }

}
